### PR TITLE
Admin REST API with client_secret

### PR DESCRIPTION
**Pull request by danifr**
Authenticating against Admin REST API with `client_id` and `client_secret`

### CODE REVIEW

File: server_development/topics/admin-rest-api.adoc [hunk 1]
```diff
@@ -10,7 +10,32 @@ A token can be obtained by enabling authenticating to your application with {pro
 
 For complete documentation see {apidocs_link}[{apidocs_name}].
 
-=== Example using CURL
+=== Examples using CURL
+
+==== Authenticating with `client_id` and `client_secret`
```
Comment (stianst) [inline server_development/topics/admin-rest-api.adoc]: I'd change this to "Authenticate with a service account" and move it below the "Authenticate with username and password" example

**Comment by danifr** [comment]
Shall we merge this? There are a few questions on StackOverflow about this topic and it would be nice if there is a solution from the official docs

File: server_development/topics/admin-rest-api.adoc [hunk 2]
```diff
@@ -37,6 +39,29 @@ curl \
...
 ----
 
+==== Authenticate with a service account
+
+Before being able to authenticate against the Admin REST API using a `client_id` and a `client_secret` you need to make sure the client is configured as it follows:
+
+* `client_id` is a **confidential** client that belongs to the realm *master*
+* `client_id` has `Service Accounts Enabled` option enabled
+* `client_id` has a custom "Hardcoded claim" mapper
+  ** Token Claim Name: `azp`
```
Comment (stianst) [inline server_development/topics/admin-rest-api.adoc]: Setting the azp here is not valid, and is misleading. The client should have scope on the required roles within security-admin-console, limited to what it needs. Then it should have the 'audience resolve' mapper added so the token has an audience on security-admin-console. which will be added automatically as it as a scope on the roles for that client.

**Review by stianst — CHANGES_REQUESTED** [review]

File: server_development/topics/admin-rest-api.adoc (hunk 2 above)
Comment (danifr) [reply server_development/topics/admin-rest-api.adoc]: Hi Stian,

thanks for your comment. I will fix it.

>  The client should have scope on the required roles within security-admin-console, limited to what it needs

Can you clarify what you mean by this? I am not sure how I can translate that sentence into keycloak config (I am trying to reproduce what you say so I can eventually document it in plain english)

Comment (stianst) [reply server_development/topics/admin-rest-api.adoc]: The steps should create a new client (not re-use admin-client client), remove full scope allowed, and select what roles it needs from the security-admin-console, then it should have the audience resolve mapper added. That will mean it has access to the roles it needs + the correct audience.

Comment (danifr) [reply server_development/topics/admin-rest-api.adoc]: Thanks for your comments. I just updated my PR.

**Comment by danifr** [comment]
Build is not passing. This commit seems to fix it though https://github.com/keycloak/keycloak-documentation/pull/918/commits/a48c7f10a2c199f5bac716059d306369d2a53a01

I will rebase from master after it gets merged.

**Review by stianst — APPROVED** [review]
